pkg/master/controller.go

```diff
@@ -155,15 +156,18 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
 			Labels:    map[string]string{"provider": "kubernetes", "component": "apiserver"},
 		},
 		Spec: api.ServiceSpec{
-			Ports: []api.ServicePort{{Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}},
+			Ports: []api.ServicePort{{
+				Protocol:   api.ProtocolTCP,
+				Port:       servicePort,
+				TargetPort: util.NewIntOrStringFromInt(servicePort),
+				NodePort:   apiServerNodePort}},
```

Comment:
I like that idea.  I was just going to leave a comment on this line as I wanted feedback for the final implementation on wether we should still support the `ClusterIP`.

I guess your answer is **yes**, which makes sense.